### DISCUSSION TITLE
Apache Maven Surefire/Failsafe Plugin 3.0.0-M6

### DISCUSSION THREAD

**Discussion by Tibor17** [quarkusio/quarkus#24681] [Community]
Hello guys.
I would like to invite you to Vote for Surefire/Failsafe Maven Plugin 3.0.0-M6.
https://lists.apache.org/thread/60j5vr86y9o24yoj4665gy1rylb7865g
Many people are asking me when we will release the final version 3.0.0. After over one decade of development, we are in the phase where we are not able to fix some issues without reworking some code internals and that's the reason why we have continuous work in milestone versions. We are still supporting our customers and we combine this activity with internal activities. The final version 3.0.0 will break the backwards compatibility in a positive manner for user.

**Comment by famod** [comment]
I'll give M6 a try in my fork of Quarkus.

**Reply by Tibor17** [reply]
The Vote seems to be successful.

**Reply by famod** [reply]
Offline life got in the way and testing a pre-release plugin in Quarkus is not that easy.

**Comment by Tibor17** [comment]
The plugins are in Maven Central.

**Reply by famod** [reply]
Cool, thanks! I've already updated my current "commercial" project. The more instant console logging alone is worth the update!

I'll prepare a PR for Quarkus.

**Comment by Tibor17** [comment]
@famod 
Pls check the POM and the configuration`useModulePath=false` as a workaround.
http://quabr.com:8182/66637732/maven-ignoring-tests-on-build
Now the workaround can be removed.

**Reply by famod** [reply]
Thanks for the pointer. I saw something in the release notes but wasn't sure it was related.
Gonna check it out.

**Comment by famod** [comment]
Quarkus CI Devtools build seems to hitting the same `BufferOverflowException` as @yrodiere already reported here: https://issues.apache.org/jira/browse/SUREFIRE-2056

**Reply by Tibor17** [reply]
@famod 
This is bad information, really.
Can we test Quarkus with `2.22.3` as another alternative?

**Reply by famod** [reply]
Well, that would be a downgrade, given we've been promoting 3.0.0-M5 for a long time.

**Reply by Tibor17** [reply]
I understand it.
Releasing M7 several days after M6 would not be looking good either.

**Reply by Tibor17** [reply]
I am not able to test Quarkus project locally on my PC. It is quite huge project.
If 2.22.3 is ok, would it be acceptable version until we finish 3.0.0-M7?

**Reply by edeandrea** [reply]
As @famod mentioned Quarkus has been promoting 3.0.0-M5 for a long time (so have other frameworks out there). I don't speak for the Quarkus team as a whole, but personally I don't see promoting a downgrade to 2.22.3 as an option.

**Reply by edeandrea** [reply]
FYI I tested the [Quarkus Superheroes sample app](https://github.com/quarkusio/quarkus-super-heroes) with M6 and everything was fine.

**Reply by famod** [reply]
Coming back to this, in my current project we ran into another issue: https://issues.apache.org/jira/browse/SUREFIRE-2079

So, unfortunately, I cannot recommend using M6 (but YMMV).

PS: I know that this milestone thing has been discussed before and there might be good reasons for that scheme, but more often than not it cripples your ability to react quickly to severe bugs (in that there is no M6.1, only the next big M7).
The end-result will be that many users have to downgrade and there will be less feedback for all those nice improvements in M7.

It boils down to release often in smaller increments, I suppose...

Just my 2 cents.

**Comment by knutwannheden** [comment]
Since we had an issue with M5 (classpath computed incorrectly) we upgraded to M6. Luckily we haven't hit any of the mentioned M6 issues, but moving to a new M7 would feel best.

**Comment by famod** [comment]
FYI, PR upgrading to M7: https://github.com/quarkusio/quarkus/pull/25985
